// Copyright 2021 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option cc_enable_arenas = true;

package ray.rpc;

// A batch of logs with metadata and multiple log lines.
message LogBatch {
  // IP of the log publisher.
  string ip = 1;
  // Ray uses string for pid sometimes, e.g. autoscaler, gcs.
  string pid = 2;
  // Job ID in hex.
  string job_id = 3;
  // Whether this is an error output.
  bool is_error = 4;
  // Multiple lines of logs.
  repeated string lines = 5;
  // Name of the actor.
  string actor_name = 6;
  // Name of the task.
  string task_name = 7;
}
